lib/events/filesessions/filestream.go

```diff
@@ -264,6 +267,43 @@ func (h *Handler) GetUploadMetadata(s session.ID) events.UploadMetadata {
 	}
 }
 
+// ReserveUploadPart reserves an upload part.
+func (h *Handler) ReserveUploadPart(ctx context.Context, upload events.StreamUpload, partNumber int64) error {
+	// Create a buffer with the max size that a part file can have.
+	buf := make([]byte, minUploadBytes+events.MaxProtoMessageSizeBytes)
+	_, err := rand.Read(buf)
+	if err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
I wasn't sure if writing an "empty" buffer would work since I've tested with `Truncate`, and it doesn't allocate space. After some tests, it is working as expected. I'll remove this part; we can discuss this better if anything appears on the integration tests.